src/rust/src/expr/datetime.rs

```diff
@@ -29,4 +29,188 @@ impl PlRExpr {
+            .into())
+    }
+
+    fn dt_epoch_seconds(&self) -> Result<Self> {
+        Ok(self
+            .clone()
+            .inner
+            .map(
+                |s| {
+                    s.take_materialized_series()
+                        .timestamp(TimeUnit::Milliseconds)
+                        .map(|ca| Some((ca / 1000).into_column()))
+                },
+                GetOutput::from_type(DataType::Int64),
+            )
+            .into())
+    }
+
+    fn dt_with_time_unit(&self, time_unit: &str) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .dt()
+            .with_time_unit(<Wrap<TimeUnit>>::try_from(time_unit)?.0)
+            .into())
+    }
+
+    fn dt_cast_time_unit(&self, time_unit: &str) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .dt()
+            .cast_time_unit(<Wrap<TimeUnit>>::try_from(time_unit)?.0)
+            .into())
+    }
+
+    fn dt_total_days(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().total_days().into())
+    }
+    fn dt_total_hours(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().total_hours().into())
+    }
+    fn dt_total_minutes(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().total_minutes().into())
+    }
+    fn dt_total_seconds(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().total_seconds().into())
+    }
+    fn dt_total_milliseconds(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().total_milliseconds().into())
+    }
+    fn dt_total_microseconds(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().total_microseconds().into())
+    }
+    fn dt_total_nanoseconds(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().total_nanoseconds().into())
+    }
+
+    fn dt_offset_by(&self, by: &PlRExpr) -> Result<Self> {
+        Ok(self.inner.clone().dt().offset_by(by.inner.clone()).into())
+    }
+
+    fn dt_is_leap_year(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().is_leap_year().into())
+    }
+
+    fn dt_dst_offset(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().dst_offset().into())
+    }
+
+    fn dt_base_utc_offset(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().base_utc_offset().into())
+    }
+
+    fn dt_month_start(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().month_start().into())
+    }
+
+    fn dt_month_end(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().month_end().into())
+    }
+
+    fn dt_century(&self) -> Result<Self> {
+        Ok(self.inner.clone().dt().century().into())
+    }
+
+    fn dt_add_business_days(
+        &self,
+        n: &PlRExpr,
+        week_mask: LogicalSexp,
+        holidays: IntegerSexp,
+        roll: &str,
+    ) -> Result<Self> {
+        let week_mask: [bool; 7] = week_mask
+            .to_vec()
+            .try_into()
+            .unwrap_or_else(|_| !unreachable!());
```

Comment:
Maybe there's a better way for this, but it seemed simpler to check that `week_mask` is made of 7 booleans on the R side and then put `!unreachable` here

Reply:
I think that makes sense.
But can't we just use something like `.map_err(savvy::Error::from)` here to avoid panic?

Reply:
If I use it instead of the unwrap_or_else call then I have "the trait bound `savvy::Error: From<Vec<bool>>` is not satisfied"